pageserver/src/http/routes.rs

```diff
@@ -2039,6 +2040,34 @@ async fn timeline_cancel_compact_handler(
     .await
 }
 
+// Get compact info of a timeline
+async fn timeline_compact_info_handler(
+    request: Request<Body>,
+    _cancel: CancellationToken,
+) -> Result<Response<Body>, ApiError> {
+    let tenant_shard_id: TenantShardId = parse_request_param(&request, "tenant_shard_id")?;
+    let timeline_id: TimelineId = parse_request_param(&request, "timeline_id")?;
+    check_permission(&request, Some(tenant_shard_id.tenant_id))?;
+    let state = get_state(&request);
+    async {
+        let tenant = state
+            .tenant_manager
+            .get_attached_tenant_shard(tenant_shard_id)?;
+        let res = tenant.get_scheduled_compaction_tasks(timeline_id);
+        let mut resp = Vec::new();
+        for item in res {
+            resp.push(json!({
+                "compact_key_range": item.compact_key_range,
```

Comment:
We don't want uses of the `json` macro in this file. I think it shouldn't be hard to declare a struct and add it to where we store the other pageserver API structs.

Reply:
resolved in https://github.com/neondatabase/neon/pull/10173/commits/f042deff091b763aa2d909b8739afebc8292ae71